plugins/hls-cabal-plugin/hls-cabal-plugin.cabal

```diff
@@ -0,0 +1,66 @@
+cabal-version:      3.0
+name:               hls-cabal-plugin
+version:            0.1.0.0
+synopsis:           Cabal integration plugin with Haskell Language Server
+description:
+  Please see the README on GitHub at <https://github.com/haskell/haskell-language-server#readme>
+
+homepage:
+license:            MIT
+license-file:       LICENSE
+author:             Fendor
+maintainer:         fendor@posteo.de
+category:           Development
+extra-source-files:
+  CHANGELOG.md
+  test/testdata/invalid.cabal
+  test/testdata/licenseCodeAction.cabal
+  test/testdata/simple.cabal
+
+common warnings
+  ghc-options: -Wall
+
+library
+  import:           warnings
+  exposed-modules:
+    Ide.Plugin.Cabal
+    Ide.Plugin.Cabal.Diagnostics
+    Ide.Plugin.Cabal.LicenseSuggest
+    Ide.Plugin.Cabal.Parse
+
+  build-depends:
+    , base                  >=4.12     && <5
+    , bytestring
+    , Cabal                 ^>=3.2 || ^>=3.4 || ^>=3.6
```

Comment:
iirc, it would not be a problem to use the same cabal version for all GHC versions here, since Cabal 3.6 still compiles with GHC 8.8. However, stack makes it difficult, as we would have to either allow newer Cabal versions for all dependencies, or make the old resolver use the newer version for virtually all dependencies (e.g. packages that needed only version revisions). Don't know what do exactly do here, however, pre-built executables aren't build with stack so it should always use Cabal 3.6, afaict.

Reply:
I think this generally isn't too much of a problem: this plugin mostly operates on cabal files, and different versions of cabal usually treat cabal files the same; differences in behaviour are controlled with the `cabal-version` field. So I think we're okay. BUT worth a comment :)